source/server/active_tcp_socket.cc

```diff
@@ -75,6 +74,29 @@ void ActiveTcpSocket::continueFilterChain(bool success) {
     bool no_error = true;
     if (iter_ == accept_filters_.end()) {
       iter_ = accept_filters_.begin();
+      if (listener_filter_max_read_bytes_ > 0) {
+        listener_filter_buffer_ = std::make_unique<Network::ListenerFilterBufferImpl>(
+            socket_->ioHandle(), listener_.dispatcher(),
+            [this](bool error) {
+              socket_->ioHandle().close();
+              error ? listener_.stats_.downstream_peek_error_.inc()
+                    : listener_.stats_.downstream_peek_remote_close_.inc();
+              continueFilterChain(false);
+            },
+            [this](Network::ListenerFilterBuffer& filter_buffer) {
+              ASSERT((*iter_)->maxReadBytes() > 0);
```

Comment:
What if there are two listener filters in the chain, one of which has maxReadBytes() > 0, and one with maxReadBytes() == 0? Is there logic somewhere else to avoid tripping this assertion?

Reply:
If the listener's maxReadBytes() == 0, then it shouldn't return `FilterStatus::StopIteration` on `onAccept()`.
https://github.com/envoyproxy/envoy/pull/17395/files/ec80e8825245aa0ed651ec63c587aa77e3d53c40#diff-6708370e112a5097f06e16009c3730b34f5ab0ee4f1e736f8816848409435888R106

Otherwise it should be a bug for the filter implementation, here is another assertion stop that https://github.com/envoyproxy/envoy/pull/17395/files/ec80e8825245aa0ed651ec63c587aa77e3d53c40#diff-6708370e112a5097f06e16009c3730b34f5ab0ee4f1e736f8816848409435888R116